pkg/cache/clusterqueue.go

```diff
@@ -297,6 +300,25 @@ func (c *clusterQueue) inactiveReason() (string, string) {
 			messages = append(messages, fmt.Sprintf("AdmissionCheck(s): %v cannot be set at flavor level", c.flavorIndependentAdmissionCheckAppliedPerFlavor))
 		}
 
+		if features.Enabled(features.TopologyAwareScheduling) && len(c.tasFlavors) > 0 {
+			if c.HasParent() {
+				reasons = append(reasons, kueue.ClusterQueueActiveReasonNotSupportedWithTopologyAwareScheduling)
+				messages = append(messages, "TAS is not supported for cohorts")
+			}
+			if c.Preemption.WithinClusterQueue != kueue.PreemptionPolicyNever {
+				reasons = append(reasons, kueue.ClusterQueueActiveReasonNotSupportedWithTopologyAwareScheduling)
+				messages = append(messages, "TAS is not supported for preemption within cluster queue")
+			}
+			if len(c.multiKueueAdmissionChecks) > 0 {
+				reasons = append(reasons, kueue.ClusterQueueActiveReasonNotSupportedWithTopologyAwareScheduling)
+				messages = append(messages, "TAS is not supported with MultiKueue admission check")
+			}
+			if len(c.provisioningAdmissionChecks) > 0 {
+				reasons = append(reasons, kueue.ClusterQueueActiveReasonNotSupportedWithTopologyAwareScheduling)
+				messages = append(messages, "TAS is not supported with ProvisioningRequest admission check")
+			}
+		}
+
```

Comment:
Could you expand this unit testing?
https://github.com/kubernetes-sigs/kueue/blob/5203ac9eb69226a1ae3b6ba6d361eaed7ef3dc8c/pkg/cache/clusterqueue_test.go#L487

Or, we may want to prepare the dedicated `TestClusterQueueUpdateWithTAS`.

Reply:
The selected code actually comes from the `inactiveReason` function. My idea is to write a unit test for this function, using `updateClusterQueue` as setup. However, I would prefer this as follow up to focus on completing the functional PRs, the integration tests we have cover all the cases here.

Reply:
> However, I would prefer this as follow up to focus on completing the functional PRs, the integration tests we have cover all the cases here.

That makes sense. Let's update the spread sheet.

Reply:
Added as a new "Add unit test for CQ validation for unsupported configurations (the inactiveReason)" entry

Reply:
Thanks.